evap/student/forms.py

```diff
@@ -80,7 +80,8 @@ def __init__(self, *args, contribution, questionnaire, **kwargs):
                 field = TextAnswerField.from_question(question)
             elif question.is_rating_question:
                 field = RatingAnswerField.from_question(question)
-            elif question.is_heading_question:
+            else:
+                assert question.is_heading_question
```

Comment:
I wonder if we could construct a richer `QuestionType` type which we could `match` against and also add an `assert_never` somewhere here - I don't like SAT solving in my head :D

Reply:
As someone without any experience of match semantics, it seems to me that the (sub)structural matching logic currently isn't able for us to have something like that without changing `QuestionType`? We kind of have the relevant (sub)structures in `QUESTION_TYPES`, but I wouldn't want to match against any translated strings here.

Do you have a specific idea how you'd want this to be implemented? I generally like the idea, but I'd say it's a separate thing for now.

Reply:
It would likely require defining a new type hierarchy like
```python3
GradeQuestion = Literal[...] | Literal[...] | ...

# or

class GradeQuestion(Enum):
    ...

QuestionType = TextQuestion | GradeQuestion | ...
```

I think this hierarchy would live outside of the database though, which could give the ORM naming problem we found with `Evaluation.state` (vs `state_`) again.

Anyways, I think we would then be able to match at whichever granularity we want, for example
```python3
match question_type:
    case TextQuestion():
        pass
    case GradeQuestion.FEW_MANY:
        pass
# ...
```

This is just a thought though, not something for this PR

Reply:
Actually, reviewing another PR I found https://stackoverflow.com/a/77236235/13679671 which lowers my certainty that this is possible, I would say ~65%